### ISSUE TITLE
Add ability to query version variables/properties using nbgv

### ISSUE THREAD

**Issue by ekwus** [AArnott/Nerdbank.GitVersioning#200]
For cloud systems where it is difficult to pass back the generated version information for further build steps, add the ability to query for a particular version property such as "NuGetPackageVersion".

Writing the value out to the console then allows platform specific script to pipe the value in to an environment variable, file or other command.

I have implemented this in my fork here;

https://github.com/ekwus/Nerdbank.GitVersioning/tree/feature/nbgv_query_values

The nbgv get-version command now supports any VersionOracle propety as the --format argument and will return the resulting value to the console.

Potentially, security of the change could be improved by adding a check against allowed properties instead of just allowing all VersionOracle public properties to be queried.

Happy to turn this in to a Pull Request or rework it to fit with your thinking. Possibly it should be an additional command completely.

Cheers

Dave

**Comment by AArnott** [comment]
If you can use powershell or otherwise parse xml or Json output this can already be done, since it can print out those formats with all the supported properties. Is there some reason this doesn't work for you?

**Comment by ekwus** [comment]
Hey

The build is currently running in a command prompt shell. I did consider parsing the json output but it means we'd need to introduce another tool to the build (of course I might be unaware of any built-in tool). Ultimately the more complexity in the scripting means more chance for issues and for others devs to misunderstand it.

I assume powershell gives you the ability to parse json/xml easily? I'm not overly familiar with it. We could switch this build to using powershell, however we are looking to use one solution across our windows/linux and C#/C++/Installers, etc.

I've got it working with an internal build of the fork so not essential to have it in the main release if it doesn't help anyone else.

Cheers

Dave
